server/src/main/java/org/elasticsearch/common/Rounding.java

```diff
@@ -1119,12 +1118,16 @@ public long round(long utcMillis) {
             }
 
             @Override
-            public long nextRoundingValue(long time) {
-                int offsetSeconds = timeZone.getRules().getOffset(Instant.ofEpochMilli(time)).getTotalSeconds();
-                long millis = time + interval + offsetSeconds * 1000;
-                return ZonedDateTime.ofInstant(Instant.ofEpochMilli(millis), ZoneOffset.UTC)
-                    .withZoneSameLocal(timeZone)
-                    .toInstant().toEpochMilli();
+            public long nextRoundingValue(long utcMillis) {
+                long from = utcMillis + interval; // TODO this implementation makes me very upset
```

Comment:
So I'm just glancing at this, and haven't read the whole class, but, it looks like the old implementation was using `interval` as a number of seconds, and this is using it as a number of milliseconds?  Is that correct?  Or am I misunderstanding something here?

Reply:
The old implementation had the zone's offset in seconds but the interval was in millis.